src/Login/VerificationLandingPage.tsx

```diff
@@ -1,60 +1,97 @@
-import { Jumbotron } from 'react-bootstrap';
+import { Button, Grid, TextField } from '@material-ui/core';
+import React from 'react';
+import { useForm } from 'react-hook-form';
 import { useParams } from 'react-router';
-import { Link } from 'react-router-dom';
+import { useHistory } from 'react-router-dom';
 import BackendAPIError from '../APIInterfaces/BackendAPI/BackendAPIError';
 import { getVerification } from '../APIInterfaces/BackendAPI/Requests/UserRequests';
 import logger from '../Utilities/Logger';
+import { Alert as MUIAlert } from '@material-ui/lab';
+import { useMUIAlertState } from '../Hooks/useAlertState';
+import LandingPageWrapper from '../Components/LandingPageWrapper';
 
 interface VerificationLandingPageProps {
 
 }
 
+interface ConfirmVerifyFormInputs {
+    confirmEmail: string;
+}
+
 // TODO: Use Axios.Request JSX to selectively render success or failure.
 export const VerificationLandingPage: React.FC<VerificationLandingPageProps> = () => {
     const { uid } = useParams<{
         uid: string;
     }>();
-    const [{verifyData, verifyError}, setVerifyState] = useState({verifyData: '', verifyError: ''});
     
-    // Async functions return promises regardlessly, and this angers Typescript. IIFE is the workaround.
-    useEffect(() => {
-        if (!uid) return;
-        (async () => {
-            try {
-                const res = await getVerification({
-                    verifyToken: uid
-                });
-                if (res.status !== 200) {
-                    logger.warn(`Verification succeeded but got a non 200 status code: ${res.status}`);
-                }
-                setVerifyState({verifyData: 'Success', verifyError: ''});
-            } catch (e) {
-                if (e instanceof BackendAPIError && e.status === 400) {
-                    setVerifyState({verifyError: 'This verification link is invalid.', verifyData: ''});                    
-                } else {
-                    logger.error('VerificationLandingPage: expected error to be a bad request but got some other error', e);
-                    setVerifyState({verifyError: e.message, verifyData: ''});
-                }
+    const history = useHistory();
+    const { register, handleSubmit } = useForm<ConfirmVerifyFormInputs>();
+    const [{ message, severity }, setUpdateAlert] = useMUIAlertState();
+
+    const onSubmit = async (data: ConfirmVerifyFormInputs) => {
+        try {
+            const res = await getVerification({
+                verifyToken: uid,
+                confirmEmail: data.confirmEmail,
+            });
+            if (res.status !== 200) {
+                logger.warn(`Verification succeeded but got a non 200 status code: ${res.status}`);
+            }
+            setUpdateAlert({message: 'Success! Your account has been verified. You will be redirected to the login page momentarily.', severity: 'success'});
+            setTimeout(()=>{
+                history.replace('/common/courses');
```

Comment:
they are presumably not logged in so this will automatically redirect to '/', we should probably just change that to /